app/Http/Controllers/OrderTrackingController.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Order;
+use Illuminate\Http\Request;
+
+class OrderTrackingController extends Controller
+{
+    public function index(){
+        $orders = Order::with('orderLine', 'orderStatus')->where('user_id', auth()->id())->get();
+
+        //Calculate most expensive product for each order and add to Associative array
+        $mostExpensiveProductsByOrder = [];
+
+        foreach ($orders as $order) {
+            $mostExpensiveProduct = null;
+
+            foreach ($order->orderLine as $orderLine) {
+                // If no product is set or the current product's price is higher than the one stored
+                if ($mostExpensiveProduct === null || $mostExpensiveProduct->price < $orderLine->product->price) {
+                    $mostExpensiveProduct = $orderLine->product;
+                }
+            }
+            // Only add to the result if there's a product (in case there are orders with no order lines)
+            if ($mostExpensiveProduct !== null) {
+                $mostExpensiveProductsByOrder[$order->id] = $mostExpensiveProduct;
+            }
+        }
+        return view('track_orders.trackOrders', ['orders' => $orders, 'mostExpensiveProductsByOrder' => $mostExpensiveProductsByOrder]);
+    }
+
+    public function details(Order $order){
+
+
+        //calculate total cost of the order
```

Comment:
Hetzelfde hiervoor, als je in de Order model een functie maakt die de totale prijs teruggeeft hoef je dat hier niet te doen